test/python/fx_importer/basic_test.py

```diff
@@ -108,6 +113,12 @@ def forward(self, x):
 @run
 # CHECK-LABEL: test_broadcast_with_dynamic_shapes
 # CHECK:     func.func @test_net(%[[ARG0:[a-zA-Z0-9]+]]: !torch.vtensor<[1,2],f32>, %[[ARG1:[a-zA-Z0-9]+]]: !torch.vtensor<[?],f32>) -> !torch.vtensor<[?,2],f32>
+# CHECK:     %[[S0:.*]] = torch.symbolic_int "s0" {min_val = {{[0-9]+}}, max_val = {{[0-9]+}}} : !torch.int
+# CHECK:     torch.bind_symbolic_shape %[[ARG1]], [%[[S0]]], affine_map<()[s0] -> (s0)> : !torch.vtensor<[?],f32>
+# CHECK:     torch.aten.size.int
+# CHECK:     torch.prim.ListConstruct
+# CHECK:     %[[EXPAND:.*]] = torch.aten.expand
+# CHECK:     torch.bind_symbolic_shape %[[EXPAND]], [%[[S0]]], affine_map<()[s0] -> (s0, 2)> : !torch.vtensor<[?,2],f32>
 def test_broadcast_with_dynamic_shapes():
     class Basic(nn.Module):
         def __init__(self):
```

Comment:
Would it be possible to add a couple of more examples of broadcasting, especially the potential multiple options around binary broadcasting against a non-unit dim e.g. tensor<3x3xf32> vs tensor<?x3xf32> or where both tensors have unknown dim shapes ? 

Reduction ops would also be a great example to see, especially with multiple unknown axial dims with keepdim=False.

Reply:
I know I've seen more complicated cases in the wild. They tend to come up when symbols are related in program arguments by some non trivial expression. Struggling to come up with one now.

Reply:
We have a collection of more complex cases but primarily within the TF ecosystem. We're keen on seeing how things look in the torch world for at least some of the simpler forms, but incremental support for these constructs is fine - I don't want to block @sjain-stanford in the quest of immediate perfection!

Reply:
Haha, no I'd be happy to include more cases to the testsuite, to help uncover rough edges. @sjarus LMK if you have any pytorch snippets you'd like me to try (either in this PR or as a follow-on), and I'll be sure to include.

Reply:
With an eye to add support for more complicated cases as we come across them in the future, marking as resolved.

Reply:
Sorry I didn't see the followon comment in time! Doing it in follow-on PR sounds reasonable at this point!